src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -471,31 +576,49 @@ public void checkResponse (String header, String method, String uri)
         }
     }
 
+    private String computeUserhash(String algorithm, String user, String realm)
+        throws NoSuchAlgorithmException
+    {
+        boolean truncate256 = false;
+
+        if (algorithm.equals("SHA-512-256")) {
```

Comment:
It appears that the incoming param `algorithm` appears can be of any case. In some other places like the `validateAlgorithm`, this `algorithm` value has been first converted to an uppercase value and then used for additional checks. Should a similar upper case conversion be done here before this equality check? Perhaps, we should convert this to upper case once and then pass it around to these `validateAlgorithm` and `computeUserhash` methods?

Reply:
Right. I'll check all that for the next round.